core/amber/src/main/python/core/models/operator.py

```diff
@@ -96,6 +96,72 @@ def on_finish(self, port: int) -> Iterator[Optional[TupleLike]]:
         yield
 
 
+class BatchOperator(TupleOperatorV2):
+    """
+    Base class for batch-oriented operators. A concrete implementation must
+    be provided upon using.
+    """
+
+    BATCH_SIZE: int = 10  # must be a positive integer
+
+    def __init__(self):
+        super().__init__()
+        self.__batch_data: MutableMapping[int, List[Tuple]] = defaultdict(list)
+        self._check_batch_size(self.BATCH_SIZE)
+
+    @staticmethod
+    def _check_batch_size(value):
```

Comment:
let's call it `_validate_batch_size` and add doc-string to mention that this method is not to be used by the any user.

Reply:
Also we do not want users to override this function, is it necessary to add `@overrides.final` guard?

Reply:
I think both works, probably should be consistent with on_finish.